src/cmd/common/common.go

```diff
@@ -0,0 +1,85 @@
+package common
+
+import (
+	"flag"
+	"fmt"
+	"log/slog"
+	"os"
+	"registry-stable/internal/github"
+	"registry-stable/internal/module"
+	"registry-stable/internal/provider"
+)
+
+type CLI struct {
+	Logger          *slog.Logger
+	Github          github.Client
+	ModuleDataDir   string
+	ProviderDataDir string
+}
+
+func Parse() CLI {
+	moduleDataDir := flag.String("module-data", "../modules", "Directory containing the module data")
+	providerDataDir := flag.String("provider-data", "../providers", "Directory containing the provider data")
+
+	flag.Parse()
+
+	cli := CLI{
+		Logger:          slog.New(slog.NewJSONHandler(os.Stdout, nil)),
+		ModuleDataDir:   *moduleDataDir,
+		ProviderDataDir: *providerDataDir,
+	}
+
+	return cli
+}
+
+func (cli CLI) Modules(action func(m module.Module) error) error {
```

Comment:
Whilst we have 2 commands now that iterate over modules and providers, I feel like this method isn't really needed. Especially as not every CLI tool we make is going to use this.

For example, We have 3-5 tools planned to look at just GPG keys, and they wont use these at all.

Reply:
I wouldn't be against a method similar to this in the module/provider packages though. Something like

`modules.ForEachModule(modulePath string, action func(m module.Module)....`